Products/Zuul/facades/metricfacade.py

```diff
@@ -51,24 +53,144 @@ def _isRunningFromUI(context):
     return safe_hasattr(context.REQUEST, 'SESSION')
 
 
+class MetricConnection(object):
+
+    """Manages communication to Metric Server."""
+
+    def __init__(self, auth_token, credentials, global_credentials):
+        """Metric connection constructor.
+
+        :param auth_token: ZAuthToken used for authentication
+        :param credentials: current user credentials
+        :param global_credentials: global user credentials
+        """
+        self._metric_url = getGlobalConfiguration().get('metric-url',
+                                                        'http://localhost:8080/')
+
+        self._auth_token = auth_token
+        self._credentials = credentials
+        self._global_credentials = global_credentials
+
+        self._req_session = self._init_session()
+
+    def _init_session(self):
+        req_session = requests.Session()
+
+        agent_suffix = 'python'
+        if sys.argv[0]:
+            agent_suffix = os.path.basename(sys.argv[0].rstrip(".py"))
```

Comment:
For the agent string, please make 'python' the default, but the override should be passed into the `__init__` method as an optional value (i.e. a keyword argument with a default value).  This should make the `MetricConnection` object more general.